File: server/internal/posts/internal/domain/posts.go
package domain

import (
	"time"

	"github.com/samber/lo"
	"go.mongodb.org/mongo-driver/v2/bson"
)

type Posts struct {
	ID          bson.ObjectID `bson:"_id"`
	CreatedAt   time.Time     `bson:"created_at"`
	UpdatedAt   time.Time     `bson:"updated_at"`
	DeletedAt   time.Time     `bson:"deleted_at,omitempty"`
	Title       string
	Content     string
	Author      string
	Description string
	Category    string
	Tags        []string
	Cover       string
	IsTop       bool `bson:"is_top"`
	Status      *int 
}

func ToPtr(posts []Posts) []*Posts {
	return lo.Map(posts, func(item Posts, _ int) *Posts {
		return &item
	})
}

File: web/src/types/posts.ts
export interface PostVO {
  id: string;
  created_at: string;
  title: string;
  content: string;
  description: string;
  author: string;
  category: string;
  tags: string[];
  cover: string;
  is_top: boolean;
  is_publish: boolean;
}

export interface IPostCard {
  id: string;
  title: string;
  content: string;
  description: string;
  created_at: string;
  author: string;
  cover?: string;
  category?: string;
  tags?: string[];
}
